cmd/validate.go

```diff
@@ -292,7 +305,7 @@ func processEntry(ctx context.Context, pw progress.Writer, wp cmdutil.Pool, e pl
 				ccrit := newCriteria(entry)
 				// If we have a TypeID, only explore children if they are different from the parent.
 				// This prevents simple recursion like volume directories containing more dirs.
-				if ccrit.typeID == "" || ccrit != crit {
+				if ccrit.typeID == "" || ccrit.typeID != crit.typeID {
```

Comment:
Wonder if this should be `ccrit.typeID == "" || (crit.typeID != "" && ccrit.typeID != crit.typeID)`